backend/src/layer/data/database/request.rs

```diff
@@ -0,0 +1,649 @@
+                    price_pupil: u32::try_from(m.price_pupil)?
+                },
+                frequency: u32::try_from(null_error!(m.frequency))?,
+                new: null_error!(m.new),
+                last_served: m.last_served,
+                next_served: m.next_served,
+                average_rating: null_error!(m.average_rating),
+                rating_count: u32::try_from(null_error!(m.rating_count))?,
+            })
+        }).transpose()
+    }
+
+    async fn get_meals(&self, line_id: Uuid, date: Date) -> Result<Option<Vec<Meal>>> {
+        // If date too far into the future, return `None`.
+        // This should probably be inside the logic layer which currently does not exists for request.
+        let today = Local::now().date_naive();
+        let age = today - date;
+        if age.num_weeks() > MAX_WEEKS_DATA {
+            return Ok(None);
+        }
+
+        sqlx::query!(
+            r#"
+            SELECT food_id, name, food_type as "meal_type: MealType",
+                price_student, price_employee, price_guest, price_pupil, serve_date as date, line_id,
+                new, frequency, last_served, next_served, average_rating, rating_count
+            FROM meal_detail JOIN food_plan USING (food_id)
+            WHERE line_id = $1 AND serve_date = $2
+            ORDER BY food_id
+            "#,
+            line_id,
+            date
+        )
+        .fetch_all(&self.pool)
+        .await?
+        .into_iter()
+        .map(|m| {
+            Ok(Meal {
+                id: null_error!(m.food_id),
+                line_id: m.line_id,
+                date: m.date,
+                name: null_error!(m.name),
+                meal_type: null_error!(m.meal_type),
+                price: Price {
+                    price_student: u32::try_from(m.price_student)?,
+                    price_employee: u32::try_from(m.price_employee)?,
+                    price_guest: u32::try_from(m.price_guest)?,
+                    price_pupil: u32::try_from(m.price_pupil)?
+                },
+                frequency: u32::try_from(null_error!(m.frequency))?,
+                new: null_error!(m.new),
+                last_served: m.last_served,
+                next_served: m.next_served,
+                average_rating: null_error!(m.average_rating),
+                rating_count: u32::try_from(null_error!(m.rating_count))?,
+            })
+        })
+        .collect::<Result<Vec<_>>>().map(Some)
+    }
+
+    async fn get_sides(&self, line_id: Uuid, date: Date) -> Result<Vec<Side>> {
+        sqlx::query!(
+            r#"
+            SELECT food_id, name, food_type as "meal_type: MealType", 
+            price_student, price_employee, price_guest, price_pupil
+            FROM food JOIN food_plan USING (food_id)
+            WHERE line_id = $1 AND serve_date = $2 AND food_id NOT IN (SELECT food_id FROM meal)
+            ORDER BY food_id
+            "#,
```

Comment:
order by uuid is random

Reply:
The ordering is (mostly) for testing purposes. The order was random before, now we have something to compare

Reply:
this ordering is exactly the opposite of random. It is deterministic but a user may not nee a pattern. This is just to avoid foods randomly switching order when the db desides to. performance whise sorting by an index should be ok (add no significant overhead)